TestRunWithDifferentArguments is not stable

It fails from time to time:
```
=== RUN   TestRunWithDifferentArguments
EOF
    cli_test.go:97: 
        	Error Trace:	cli_test.go:97
        	            				cli_test.go:246
        	Error:      	Received unexpected error:
        	            	EOF
        	Test:       	TestRunWithDifferentArguments
--- FAIL: TestRunWithDifferentArguments (0.07s)
```

--- a/pkg/compiler/compiler_test.go
+++ b/pkg/compiler/compiler_test.go
@@ -61,13 +61,13 @@ func TestCompiler(t *testing.T) {
 				require.NoError(t, err)
 				err = os.MkdirAll(exampleSavePath, os.ModePerm)
 				require.NoError(t, err)
-				outfile := exampleSavePath + "/test.nef"
-				_, err = compiler.CompileAndSave(exampleCompilePath+"/"+infos[0].Name(), &compiler.Options{Outfile: outfile})
-				require.NoError(t, err)
 				defer func() {
 					err := os.RemoveAll(exampleSavePath)
 					require.NoError(t, err)
 				}()
+				outfile := exampleSavePath + "/test.nef"
+				_, err = compiler.CompileAndSave(exampleCompilePath+"/"+infos[0].Name(), &compiler.Options{Outfile: outfile})
+				require.NoError(t, err)
 			},
 		},
 	}

--- a/pkg/vm/cli/cli_test.go
+++ b/pkg/vm/cli/cli_test.go
@@ -75,16 +75,16 @@ func newTestVMCLIWithLogo(t *testing.T, printLogo bool) *executor {
 			Stdin:  e.in,
 			Stdout: e.out,
 		})
-	go func() {
-		require.NoError(t, e.cli.Run())
-		close(e.ch)
-	}()
 	return e
 }
 
 func (e *executor) runProg(t *testing.T, commands ...string) {
 	cmd := strings.Join(commands, "\n") + "\n"
 	e.in.WriteString(cmd + "\n")
+	go func() {
+		require.NoError(t, e.cli.Run())
+		close(e.ch)
+	}()
 	select {
 	case <-e.ch:
 	case <-time.After(time.Second):

